rikai/app/Helpers/rate.php

```diff
@@ -49,4 +50,21 @@ function categorybook(){
 
    }
  }
+
+ function allCategorybook(){
+   $categorys = Category::where('parent_id','=','0')->get();
+
+   foreach($categorys as $category){
+      echo  '<li class="dropdown">';
+      echo '<a href="'.route('categoryuser.edit',[$category->id]).'">'.__('message.'.$category->title).'<span class="expand">&raquo;</span>'.'</a>';
+      echo  '<ul class="child">';
+         foreach($category->subcategory()->get() as $cat){
+            echo  '<li class="dropdown">';
+            echo  '<a href="'.route('categoryuser.edit',[$cat->id]).'" nowrap>'.__('message.'.$cat->title).'</a><br>';
+            echo  '</li>';
+         }
+      echo  '</ul>';
+      echo  '</li>';
+   }
+ }
```

Comment:
Đưa ra view nhé.
Trường hợp nào ngắn thì có thể đưa vào helper

Reply:
Với lại sao lại xử lý echo từng dòng như thế này.
Gộp nó lại 1 cục rồi return 1 lần luôn chứ.

Reply:
em đã fix bằng cách cho nó ra ngoài view, còn cái echo lúc đầu do nó nối chuối và in ra dữ liệu bình thường nên em làm theo kiểu echo sẽ nhanh hơn , bây h em không cần hàm đó nữa nên em xóa rồi ạ